test/distributed/test_c10d_spawn.py

```diff
@@ -241,11 +241,7 @@ def _test_base(self, net, inp, check_allclose=True):
         store = c10d.FileStore(self.file.name, self.world_size)
         process_group = c10d.ProcessGroupGloo(store, self.rank, self.world_size)
         if inp[0].is_cuda:
-            num_gpus = torch.cuda.device_count()
-            batch_size = inp[0].size(0)
-            # batch_size must be evenly divisible by num_gpus_used, take the largest one
-            num_gpus_used = [i for i in range(1, num_gpus + 1) if batch_size % i == 0][-1]
-            device_ids = list(range(num_gpus_used))
+            device_ids = [torch.cuda.current_device()]
```

Comment:
We need a comprehensive set of tests that validate our new API contract, a few examples:

1. Specifying multiple device_ids results in an error.
2. Specifying device_ids None with a single device model works fine and uses the current device.
3. Specifying device_ids None with a multi device model works fine too.

Reply:
Thanks for the suggestion! Added 3 test cases in test_c10d.py